core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -796,6 +796,23 @@ object IO {
     in.foldLeft(point[E, mutable.Builder[B, M[B]]](cbf(in)))((io, b) => io.zipWith(fn(b))(_ += _))
       .map(_.result())
 
+  /**
+   * Evaluate the elements of a traversable data structure in parallel
+   * and collect the results.
+   *
+   * _Note_: ordering in the input collection is not preserved
+   */
+  def parTraverse[E, A, B, M[X] <: TraversableOnce[X]](
```

Comment:
I know this is probably going to sound too far-fetched, but there is an issue with partial unification and non-trivial `Traversable` children like `Map`: https://users.scala-lang.org/t/problem-with-partial-unification-and-map-traversal/2083.

There is a workaround for a similar issue in e.g. scalatest: https://github.com/scalatest/scalatest/pull/1375/files. I'm just wondering if you would consider doing something similar here - that is, if someone besides me thinks this is actually an issue.

Reply:
imho It looks like a bad design choice in the scala collection library ([one of many](https://www.reddit.com/r/haskell/comments/1pjjy5/odersky_the_trouble_with_types_strange_loop_2013/cd3bgcu/)) that should be addressed in a wholistic fashion, i.e. via proper `Traversable`s and `Applicative`s.

Reply:
> i.e. via proper Traversables and Applicatives.

Don't think this is happening scala/bug#10934  :smile:

Reply:
I'm fine with the workaround, but if this is just for `Map`, I'm not sure it's worth it. In any case, let's do it in a followup PR.